lib/engine-dylib/src/artifact.rs

```diff
@@ -58,6 +58,12 @@ pub struct DylibArtifact {
     frame_info_registration: Mutex<Option<GlobalFrameInfoRegistration>>,
 }
 
+impl Drop for DylibArtifact {
+    fn drop(&mut self) {
+        std::fs::remove_file(&self.dylib_path).expect("cannot delete the temporary artifact");
```

Comment:
This will drop the file always, however there is only one scenario where we should do it.

`deserialize` (which creates a temporal file and then calls`deserialize_from_file_unchecked`) should delete the temporal file, however `deserialize_from_file` (which doesn't create a temporal file) should *not* delete the file (since it was provided by the user), we can provably have a new field specifying if the file is temporal or not and delete it on such case.

Reply:
Ok got it thank you, btw what's your policies regarding to error in this case ? Do you prefer to panic as I do here or just display a log ?

Reply:
I think it might be better to gracefully dismiss the error in case the worst case scenario is reached, so logging seems like a better option than a panic.

Reply:
That's what I think too. Updated, thanks :)

Reply:
That's great, will review now!